lib/charms/opensearch/v0/opensearch_locking.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Class for Setting configuration in opensearch config files."""
+import logging
+
+from charms.opensearch.v0.constants_charm import PeerRelationName
+from charms.opensearch.v0.helper_databag import Scope
+from charms.opensearch.v0.opensearch_exceptions import (
+    OpenSearchHttpError,
+    OpenSearchOpsLockAlreadyAcquiredError,
+)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "0924c6d81c604a15873ad43498cd6895"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+logger = logging.getLogger(__name__)
+
+
+class OpenSearchOpsLock:
+    """This class covers the configuration changes depending on certain actions."""
+
+    LOCK_INDEX = ".ops_lock"
+    PEER_DATA_LOCK_FLAG = "ops_removing_unit"
+
+    def __init__(self, charm):
+        self._charm = charm
+        self._opensearch = charm.opensearch
+
+    def acquire(self):
+        """Method for Acquiring the "ops" lock."""
+        # no lock acquisition needed if only 1 unit remaining
+        if len(self._charm.model.get_relation(PeerRelationName).units) == 1:
+            return
+
+        host = self._charm.unit_ip if self._opensearch.is_node_up() else None
+
+        # can use opensearch to lock
+        if host is not None or self._charm.alt_hosts:
```

Comment:
Should we try lock via OpenSearch if lock exists in peer databag?

Reply:
Good point, thank you